pkg/issuer/acme/dns/dns_test.go

```diff
@@ -371,11 +374,67 @@ func TestRoute53TrimCreds(t *testing.T) {
 	expectedR53Call := []fakeDNSProviderCall{
 		{
 			name: "route53",
-			args: []interface{}{"test_with_spaces", "AKIENDINNEWLINE", "", "us-west-2"},
+			args: []interface{}{"test_with_spaces", "AKIENDINNEWLINE", "", "us-west-2", false},
 		},
 	}
 
 	if !reflect.DeepEqual(expectedR53Call, f.DNSProviders.calls) {
 		t.Fatalf("expected %+v == %+v", expectedR53Call, f.DNSProviders.calls)
 	}
 }
+
+func TestRoute53AmbientCreds(t *testing.T) {
+	type result struct {
+		expectedCall *fakeDNSProviderCall
+		expectedErr  error
+	}
+
+	tests := []struct {
+		in  fixture
+		out result
+	}{
+		{
+			fixture{
+				Issuer: newIssuer("test", "default", []v1alpha1.ACMEIssuerDNS01Provider{
+					{
+						Name: "fake-route53",
+						Route53: &v1alpha1.ACMEIssuerDNS01ProviderRoute53{
+							Region: "us-west-2",
+						},
+					},
+				}),
+				DNSProviders: newFakeDNSProviders(),
+				Certificate: newCertificate("test", "default", "example.com", nil, []v1alpha1.ACMECertificateDomainConfig{
+					{
+						Domains: []string{"example.com"},
+						DNS01: &v1alpha1.ACMECertificateDNS01Config{
+							Provider: "fake-route53",
+						},
+					},
+				}),
+				Ambient: true,
+			},
+			result{
+				expectedCall: &fakeDNSProviderCall{
+					name: "route53",
+					args: []interface{}{"", "", "", "us-west-2", true},
+				},
+			},
+		},
```

Comment:
Can we get a corresponding fixture where ambient is set to false?

Reply:
done